### PR TITLE
Set PATH using real SDK and NDK directories

### PR DESCRIPTION
**Pull request by dbnicholson**
This is a regression from f7f8cea636714dc4843b8818c553510f99084b6a.
Previously, `self.sdk_dir` and `self.ndk_dir` were passed to
`select_and_check_toolchain_version`.

### CODE REVIEW

**Comment by misl6** [comment]
Ouch, I missed that! Agree about these changes.
What about line `374` ?
Maybe we should add a test or  a comment that cover this case?

**Comment by dbnicholson** [comment]
> Ouch, I missed that! Agree about these changes. What about line `374` ? Maybe we should add a test or a comment that cover this case?

Whoops, missed that line as I was focused on the f-strings. Let me see if I can make a test for this.

**Comment by dbnicholson** [comment]
Updated now with the missing line and a minimal test. It passed locally, but this is my first time in this test suite so I might not be doing the right thing.

**Comment by misl6** [comment]
Unfortunately the tests are failing, can you take care of it?

**Comment by dbnicholson** [comment]
Sure, I'll take a look on Monday.

**Comment by dbnicholson** [comment]
Hmm, it passes when I run only `tests/test_recipe.py`. Some other test must be altering `PATH` without cleaning up.

**Comment by dbnicholson** [comment]
Oh, I see what's going on now. This actually comes from f7f8cea636714dc4843b8818c553510f99084b6a, but my added test exercises it. Prior to that commit, `select_and_check_toolchain_version` would alter `os.environ`. After that commit, the `Context` class attribute `env` is altered.

Since the test I added runs `Context.prepare_build_environment`, the global `Context.env` class attribute is altered for any subsequent test. I can think of 3 possible fixes:

- Change `Context.prepare_build_environment` to alter `os.environ` like it used to.
- Change my new test to mock access to `Context.env` so `prepare_build_environment` doesn't affect subsequent tests
- Change `env` to be a `Context` instance attribute instead of a class attribute. I'd argue this is the most proper, but I don't know if there are other implications of it.
- Change `test_get_recipe_env_with` so it compares against `self.ctx.env['PATH']` instead of `os.environ['PATH']`. I think this change should be made regardless since that's actually what `Arch.get_env` does. Whether `self.ctx.env` and `os.environ` are the same is an orthogonal check.

**Comment by dbnicholson** [comment]
I think this should be fixed now. I changed the new test so it patches `Context.env` and also corrected `test_get_recipe_env_with` so it compares against `ctx.env` instead of `os.environ` even though it's not needed with the patching.

File: tests/test_recipe.py [hunk 1]
```diff
@@ -294,7 +294,7 @@ def test_get_recipe_env_with(
         #  check that the mocks have been called
         mock_ensure_dir.assert_called()
         mock_find_executable.assert_called_once_with(
-            expected_compiler, path=os.environ['PATH']
+            expected_compiler, path=self.ctx.env['PATH']
```
Comment (misl6) [inline tests/test_recipe.py:297]: Good catch!

File: tests/test_build.py [hunk 2]
```diff
@@ -89,3 +93,40 @@ def test_android_manifest_xml(self):
...
+            user_sdk_dir='sdk',
+            user_ndk_dir='ndk',
+            user_android_api=None,
+            user_ndk_api=None,
+        )
+
+        real_sdk_dir = os.path.join(os.getcwd(), 'sdk')
+        real_ndk_dir = os.path.join(os.getcwd(), 'ndk')
+        assert context.sdk_dir == real_sdk_dir
+        assert context.ndk_dir == real_ndk_dir
```
Comment (misl6) [inline tests/test_build.py:121]: Super-minor, can we add a comment describing why is that needed?
Is very likely that in the future we will not immediately get why that was needed.

**Review by misl6 — APPROVED** [review]
LGTM, thank you!

I've just added a minor suggestion.
I will wait to merge, so you can address it, but not consider it as a blocking :)

File: tests/test_build.py (hunk 2 above)
Comment (dbnicholson) [reply tests/test_build.py:121]: Sure.

**Comment by dbnicholson** [comment]
I added the comment, ready to review again.

**Comment by misl6** [comment]
> I added the comment, ready to review again.

Thank you!

**Review by misl6 — APPROVED** [review]

**Pull request merged**